Content.Shared/Clothing/EntitySystems/ToggleableClothingSystem.cs

```diff
@@ -51,252 +56,505 @@ public override void Initialize()
-            DebugTools.Assert(Exists(component.ClothingUid), "Toggleable clothing is missing expected entity.");
-            DebugTools.Assert(TryComp(component.ClothingUid, out AttachedClothingComponent? comp), "Toggleable clothing is missing an attached component");
-            DebugTools.Assert(comp?.AttachedUid == uid, "Toggleable clothing uid mismatch");
+            comp.ClothingPrototypes.Add(comp.Slot, comp.ClothingPrototype.Value);
         }
-        else
+
+        var xform = Transform(toggleable.Owner);
+
+        if (comp.ClothingPrototypes == null)
+            return;
+
+        var prototypes = comp.ClothingPrototypes;
+
+        foreach (var prototype in prototypes)
         {
-            var xform = Transform(uid);
-            component.ClothingUid = Spawn(component.ClothingPrototype, xform.Coordinates);
-            var attachedClothing = EnsureComp<AttachedClothingComponent>(component.ClothingUid.Value);
-            attachedClothing.AttachedUid = uid;
-            Dirty(component.ClothingUid.Value, attachedClothing);
-            _containerSystem.Insert(component.ClothingUid.Value, component.Container, containerXform: xform);
-            Dirty(uid, component);
+            var spawned = Spawn(prototype.Value, xform.Coordinates);
+            var attachedClothing = EnsureComp<AttachedClothingComponent>(spawned);
+            attachedClothing.AttachedUid = toggleable;
+            EnsureComp<ContainerManagerComponent>(spawned);
+
+            comp.ClothingUids.Add(spawned, prototype.Key);
+            _containerSystem.Insert(spawned, comp.Container, containerXform: xform);
+
+            Dirty(spawned, attachedClothing);
         }
 
-        if (_actionContainer.EnsureAction(uid, ref component.ActionEntity, out var action, component.Action))
-            _actionsSystem.SetEntityIcon(component.ActionEntity.Value, component.ClothingUid, action);
+        Dirty(toggleable, comp);
+
+        if (_actionContainer.EnsureAction(toggleable, ref comp.ActionEntity, out var action, comp.Action))
+            _actionsSystem.SetEntityIcon(comp.ActionEntity.Value, toggleable, action);
+    }
+
+    // Checks status of all attached clothings toggle status
+    public ToggleableClothingAttachedStatus GetAttachedToggleStatus(EntityUid toggleable, ToggleableClothingComponent? component = null)
+    {
+        if (!Resolve(toggleable, ref component))
+            return ToggleableClothingAttachedStatus.NoneToggled;
+
+        var container = component.Container;
+        var attachedClothings = component.ClothingUids;
+
+        // If entity don't have any attached clothings it means none toggled
+        if (container == null || attachedClothings.Count == 0)
+            return ToggleableClothingAttachedStatus.NoneToggled;
+
+        var toggledCount = 0;
+
+        foreach (var attached in attachedClothings)
+        {
+            if (container.Contains(attached.Key))
+                continue;
+
+            toggledCount++;
+        }
```

Comment:
I think this or `.Select(whatever I wrote).Count()` will work
```suggestion
        var toggledCount = attachedClothings.Count(c => container.Contains(c.Key));
```